services/src/pro/datasets/external/copernicus_dataspace/sentinel2.rs

```diff
@@ -0,0 +1,573 @@
+        .collect()
+    }
+
+    // TODO: reuse most of the logic for all copernicus products
+    async fn create_gdal_params(
+        &self,
+        item: &stac::Item,
+    ) -> Result<GdalDatasetParameters, CopernicusSentinel2Error> {
+        let asset_url = item
+            .s3_assert_product_url()
+            .context(MissingAssetProductUrl)?;
+
+        let file_path = PathBuf::from(&format!(
+            "{}:/vsis3{}/{}:{}m:EPSG_{}",
+            self.product.driver_name(),
+            asset_url,
+            self.product.main_file_name(),
+            self.band.resolution_meters(),
+            self.zone.epsg_code()
+        ));
+
+        let config_options = self.config_options();
+
+        let config_options_clone = config_options.clone();
+        let file_path_clone = file_path.clone();
+        let channel = self.band.channel_in_subdataset();
+
+        let mut gdal_params = crate::util::spawn_blocking(move || {
+            // set config options for the current thread and revert them on drop
+            let _config_options_holder =
+                TemporaryGdalThreadLocalConfigOptions::new(&config_options_clone);
+
+            let dataset_options = DatasetOptions {
+                open_flags: GdalOpenFlags::GDAL_OF_READONLY
+                    | GdalOpenFlags::GDAL_OF_RASTER
+                    | GdalOpenFlags::GDAL_OF_VERBOSE_ERROR,
+                allowed_drivers: None,
+                open_options: None,
+                sibling_files: None,
+            };
+
+            let dataset = gdal_open_dataset_ex(&file_path_clone, dataset_options)
+                .context(CouldNotOpenDataset)?;
+
+            gdal_parameters_from_dataset(&dataset, channel, &file_path, None, None)
+                .context(CouldNotGenerateGdalParameters)
+        })
+        .await
+        .context(Join)??;
+
+        // drop(config_options);
+        gdal_params.gdal_config_options = Some(config_options);
+
+        Ok(gdal_params)
+    }
+}
+
+#[async_trait]
+impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle> for Sentinel2Metadata {
+    async fn loading_info(
+        &self,
+        query: RasterQueryRectangle,
+    ) -> geoengine_operators::util::Result<GdalLoadingInfo> {
+        self.crate_loading_info(query).await.map_err(|e| {
+            geoengine_operators::error::Error::LoadingInfo {
+                source: Box::new(e),
+            }
+        })
+    }
+
+    async fn result_descriptor(&self) -> geoengine_operators::util::Result<RasterResultDescriptor> {
+        Ok(RasterResultDescriptor {
+            data_type: self.band.data_type(),
+            spatial_reference: SpatialReference::new(
+                SpatialReferenceAuthority::Epsg,
+                self.zone.epsg_code(),
+            )
+            .into(),
+            time: None, // TODO: specify time (2015, open end/current date)?
+            bbox: None, // TODO: exclude parts that are never visited by the satellite?
```

Comment:
use the UTM Zone box here

Reply:
78cc4aeb888efc047085c53ac2333eeaf1dd45de